pkg/cmd/monitor/health/health.go

```diff
@@ -0,0 +1,142 @@
+package health
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"go.etcd.io/etcd/clientv3"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/connectivity"
+)
+
+const (
+	SerializedReadSingleTarget CheckName = "SerializedReadSingleTarget"
+	QuorumReadSingleTarget     CheckName = "QuorumReadSingleTarget"
+	GRPCReadySingleTarget      CheckName = "GRPCReadySingleTarget"
+	QuorumRead                 CheckName = "QuorumRead"
+
+	DefaultSlowRequestDuration = 5 * time.Second
+	DefaultContextDeadline     = 60 * time.Second
+	DefaultNamespaceKey        = "/kubernetes.io/namespaces/default"
+)
+
+type CheckFunc func(*Check)
+
+type CheckName string
+
+type CheckTarget string
+
+type CheckStatus struct {
+	logDisruptionStart time.Time
+	logDisruptionEnd   time.Time
+	available          bool
+	disruption         bool
+	slowRequest        bool
+	check              CheckName
+}
+
+type Check struct {
+	ctx     context.Context
+	client  *clientv3.Client
+	lg      *zap.Logger
+	name    CheckName
+	targets []string
+	status  CheckStatus
+	Probe   CheckFunc
+}
+
+func NewCheck(ctx context.Context, lg *zap.Logger, client *clientv3.Client, targets []string) *Check {
+	return &Check{
+		ctx:     ctx,
+		client:  client,
+		lg:      lg,
+		targets: targets,
+	}
+}
+
+// Health checks will log requests over DefaultSlowRequestDuration with a timeout at DefaultContextDeadline.
+
+// WithSerializedReadSingleTarget performs a serialized RangeRequest against a single target.
+func WithSerializedReadSingleTarget() CheckFunc {
+	return func(c *Check) {
+		c.name = SerializedReadSingleTarget
+		c.checkKVSRead([]clientv3.OpOption{clientv3.WithSerializable(), clientv3.WithKeysOnly()})
+	}
+}
+
+// WithQuorumReadSingleTarget performs a quorate RangeRequest against a single target.
```

Comment:
This is what kube uses, correct?

Reply:
WithQuorumRead would be directly reflective of apiserver usage